scanner/src/main/kotlin/RemoteScanner.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2020 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.scanner
+
+import org.ossreviewtoolkit.model.ScannerDetails
+import org.ossreviewtoolkit.model.config.ScannerConfiguration
+
+/**
+ * Abstraction for a [Scanner] that runs on a remote host.
+ */
+abstract class RemoteScanner(name: String, config: ScannerConfiguration) : Scanner(name, config) {
+    abstract fun getVersion(): String
+
+    abstract fun getConfiguration(): String
```

Comment:
What's the reason for creating this class? These functions also exist in LocalScanner. Have you considered moving those instead to Scanner.

Apart from that, would it make sense to switch to a fully asynchronous interface already now for not blocking our threads for too long?

Reply:
> What's the reason for creating this class?

The reason is to unblock the [pending ScanOSS PR](https://github.com/oss-review-toolkit/ort/pull/3278) which currently uses the `LocalScanner` class due to a lack of alternatives, and to align the starting point of the `RemoteScanner` class with the also pending [FOSSID PR](https://github.com/oss-review-toolkit/ort/pull/3105).

> These functions also exist in LocalScanner. Have you considered moving those instead to Scanner.

No, but I was actually considering to *remove* `getConfiguration()` as I'm unsure how applicable implementing it would be for remote scanners. But IMO time will tell, and we could refactor this later.

> Apart from that, would it make sense to switch to a fully asynchronous interface already now for not blocking our threads for too long?

I would like to postpone this rather complex topic. We currently have multiple scenarios in mind how remote scanners could be used. To one extreme, we could initially use a remote scanner implementation only to *trigger* scans from an ORT pipeline, but not wait for or get results at all. Results would then need to be inspected from the web interface to the remote scanner, for example. The other extreme is that we simply block for results to be ready for download; this actually is not as extreme as it sounds as at least for the snippets scanners we are currently looking at, results would be available much quicker than for an average ScanCode run.

Reply:
> But IMO time will tell, and we could refactor this later.

Makes sense.